### PR TITLE
cloudstack_pf.py bug fixes

### PR DESCRIPTION
**Pull request by xied75**
Lots of fixes.

May need further refactoring: cloudstack API deprecated 'cidrlist' from createPortForwardingRule parameters. Although even the most recent document didn't mention the deprecation at all. http://cloudstack.apache.org/docs/api/apidocs-4.5/user/createPortForwardingRule.html

This commit simply set the default to empty string.

### CODE REVIEW

**Comment by resmo** [comment]
Thanks for the fixes! This module was not really meant to be complete. Where do I see cidrlist is deprecated?

File: cloudstack_pf.py [hunk 1]
```diff
@@ -320,16 +321,18 @@ def create_portforwarding_rule(self, portforwarding_rule):
         if not portforwarding_rule:
             self.result['changed'] = True
             args = {}
-            args['cidrlist'] = self.module.params.get('cidr')
```
Comment (resmo) [inline cloudstack_pf.py:323]: don't remove cidrlist here

Comment (xied75) [inline cloudstack_pf.py:323]: This is what you get if you have this line in the code:

```
failed: [localhost] => {"failed": true}
msg: CloudStackException: ('HTTP 431 response from CloudStack', <Response [431]>, {u'errorcode': 431, u'uuidList': [], u'cserrorcode': 4350, u'errortext': u'Parameter cidrList is deprecated; if you need to open firewall rule for the specific cidr, please refer to createFirewallRule command'})

FATAL: all hosts have already failed -- aborting
```

**Comment by resmo** [comment]
I see, then we should completely remove cidrlist from params and docs. Agree?

**Comment by xied75** [comment]
Then we are one step further than cloudstack docs. :)
Given they are still listing the param there, we should do the same I feel. But do enough change in the code to protect the code.

Maybe file a PR on cs docs also, after they remove it, we can clean up this cidrlist altogether.

**Comment by resmo** [comment]
Adding code just to show an error message now or in the future is pointless, isn't it?

**Comment by xied75** [comment]
What I mean is as long as line 323 is not there then the code will work, and parameters are compatible with cs doc. No need to remove this cidrlist totally. Feel free to adjust. :)

**Pull request merged**